prov/efa/src/rxr/rxr_pkt_type_misc.c

```diff
@@ -646,7 +646,23 @@ void rxr_pkt_handle_receipt_recv(struct rxr_ep *ep,
 		return;
 	}
 
-	rxr_cq_handle_tx_completion(ep, tx_entry);
+	if (!(tx_entry->rxr_flags & RXR_LONG_MSG)) {
+		rxr_cq_handle_tx_completion(ep, tx_entry);
+	} else {
+		/*
+		 * For long message protocol, when FI_DELIVERY_COMPLETE
+		 * is requested, we have to write tx completions
+		 * in either rxr_pkt_handle_data_send_completion()
+		 * or rxr_pkt_handle_receipt_recv() depending on which of them
+		 * is called later due to avoid accessing released
+		 * tx_entry.
+		 */
+		if (tx_entry->total_len == tx_entry->bytes_acked)
+			rxr_cq_handle_tx_completion(ep, tx_entry);
+		else
+			tx_entry->rxr_flags |= RXR_RECEIPT_RECEIVED_BUT_NOT_COMPLETED;
```

Comment:
```RXR_RECEIPT_RECEIVED_BUT_NOT_COMPLETED``` is overly complicated. Cannot we just set the flag regardless?

Reply:
Branching logic in `rxr_pkt_handle_receipt_recv` has changed